### ISSUE TITLE
Panicked when testing URL of proxy group is invalid

### ISSUE THREAD

**Issue by balthild** [Dreamacro/clash#113]
For example:

```yaml
Proxy Group:
- { name: "auto", type: url-test, proxies: ["ss1", "ss2", "vmess1"], url: "invalid_url", interval: 300 }
```

results in

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x557b18dc7265]

goroutine 17 [running, locked to thread]:
github.com/Dreamacro/clash/adapters/outbound.(*URLTest).Name(0x0, 0xc00015e3c0, 0xc000158140)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/adapters/outbound/urltest.go:32 +0x5
github.com/Dreamacro/clash/adapters/outbound.NewSelector(0x557b18ec1397, 0x6, 0xc000166300, 0x6, 0x8, 0xc000166300, 0x4, 0x8)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/adapters/outbound/selector.go:67 +0x7c
github.com/Dreamacro/clash/config.parseProxies(0xc000170000, 0xc000106300, 0x0, 0x0)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/config/config.go:302 +0x1905
github.com/Dreamacro/clash/config.Parse(0xc00014e030, 0x2f, 0x24, 0x557b18ec2b03, 0xa)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/config/config.go:124 +0xbf
github.com/Dreamacro/clash/hub/executor.ParseWithPath(0xc00014e030, 0x2f, 0x2f, 0x0, 0xc000000780)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/hub/executor/executor.go:20 +0x37
github.com/Dreamacro/clash/hub/executor.Parse(0x557b18b14945, 0xc000024118, 0x0)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/hub/executor/executor.go:15 +0x47
github.com/Dreamacro/clash/hub.Parse(0xc00014c040, 0xc000034a00)
	/home/balthild/go/pkg/mod/github.com/!dreamacro/clash@v0.10.2/hub/hub.go:10 +0x28
main.clash_hub_parse(0xc000000008)
	/home/balthild/Projects/Gtklash/subprojects/libclash/clash.go:15 +0x28
main._cgoexpwrap_4281fa2a07cc_clash_hub_parse(0x0)
	_cgo_gotypes.go:71 +0x53
```

The URL should be validated when `hub.Parse()` to avoid `nil` pointer reference.

**Comment by Dreamacro** [comment]
Got it, would fix next release.

**Comment by Dreamacro** [comment]
Fixed in 53b5ef1
